File: kata-others/kata-cpp/what_day_was_it.cpp
#include <iostream>
#include <string>

int main()
{
    std::string input;
    while (std::getline(std::cin, input))
    {
        long long time = std::stoi(input) * 86400; // seconds per day
        int day = (((time / 86400) + 4) % 7 + 7) % 7;
        const char *day_names[] = {
            "Sunday", "Monday", "Tuesday", "Wednesday",
            "Thursday", "Friday", "Saturday"};

        std::cout << day_names[day] << std::endl;
    }
    return 0;
}

File: kata-others/kata-rust/src/bin/what_day_was_it.rs
use std::io::{self, BufRead};

fn main() {
    let stdin = io::stdin();
    let weekdays = [
        "Monday",
        "Tuesday",
        "Wednesday",
        "Thursday",
        "Friday",
        "Saturday",
        "Sunday",
    ];
    // negative input as well
    for line in stdin.lock().lines() {
        let days_to_add: i64 = line.unwrap().trim().parse().unwrap_or(0);
        let result = ((days_to_add + 3) % 7 + 7) % 7;
        println!("{}", weekdays[result as usize]);
    }
}
